ui/redux/actions/subscriptions.js

```diff
@@ -5,83 +5,55 @@ import { Lbryio } from 'lbryinc';
 import { doClaimRewardType } from 'redux/actions/rewards';
 import { parseURI } from 'util/lbryURI';
 import { doAlertWaitingForSync } from 'redux/actions/app';
+import { selectIsSubscribedForUri } from 'redux/selectors/subscriptions';
 import { doToast } from 'redux/actions/notifications';
 
-type SubscriptionArgs = {
-  channelName: string,
-  uri: string,
-  notificationsDisabled?: boolean,
-};
-
-export function doToggleSubscription(
-  subscription: SubscriptionArgs,
-  followToast: boolean,
-  isSubscribed: boolean = false
-) {
+export function doToggleSubscription(subscription: Subscription, hideToast?: boolean) {
   return async (dispatch: Dispatch, getState: GetState) => {
+    const state = getState();
+
     const {
-      settings: { daemonSettings },
       sync: { prefsReady: ready },
-    } = getState();
+    } = state;
 
-    if (!ready) {
-      return dispatch(doAlertWaitingForSync());
-    }
+    if (!ready) return dispatch(doAlertWaitingForSync());
 
-    const { share_usage_data: shareSetting } = daemonSettings;
-    const isSharingData = shareSetting || IS_WEB;
+    const { uri: subscriptionUri, channelName, notificationsDisabled } = subscription;
 
-    if (!isSubscribed) {
-      const subscriptionUri = subscription.uri;
-      if (!subscriptionUri.startsWith('lbry://')) {
-        throw Error(`Subscription uris must include the "lbry://" prefix.\nTried to subscribe to ${subscriptionUri}`);
-      }
+    const isSubscribed = selectIsSubscribedForUri(state, subscriptionUri);
+    const { channelClaimId } = parseURI(subscriptionUri);
+
+    if (!isSubscribed && !subscriptionUri.startsWith('lbry://')) {
+      throw Error(`Subscription uris must include the "lbry://" prefix.\nTried to subscribe to ${subscriptionUri}`);
     }
 
     dispatch({
       type: !isSubscribed ? ACTIONS.CHANNEL_SUBSCRIBE : ACTIONS.CHANNEL_UNSUBSCRIBE,
       data: subscription,
     });
 
-    // if the user isn't sharing data, keep the subscriptions entirely in the app
-    if (isSharingData || IS_WEB) {
```

Comment:
Do we still plan to on using the `share_usage_data` setting in the future, say for certain membership types?  If yes, the removal of `isSharingData` makes it not obvious that this block needs to be skipped if we want to disable data-sharing.

I think in the first place, the `|| IS_WEB` hack is not clean. We probably could have just clamped the setting to `true` for Odysee.

Reply:
Don't think we plan to use that setting, and if we do something, it would be a different one.